src/python/pycold/imagetool/prepare_ard.py

```diff
@@ -1840,6 +1840,8 @@ def main(
         ]
     elif collection == "HLS":
         folder_list = [f for f in listdir(source_dir) if f.startswith("HLS")]
+        if folder_list == []:
+            folder_list = [folder for folder, dir, file in os.walk(source_dir) if folder.split('/')[-1].startswith('HLS.')]
```

Comment:
`folder.split('/')[-1]` can be replaced by `os.path.basename`, which will work even if the system path separator is not `/` (e.g. on windows). What I think is even more readable is using `pathlib`, which a lot of this code would benefit from:

```python
from pathlib import Path
...

folder_list = [folder for folder, dir, file in os.walk(source_dir) if Path(folder).name.startswith('HLS.')]
```